cocos/network/WebSocket.cpp

```diff
@@ -900,7 +900,7 @@ void WebSocket::onConnectionClosed()
     _readStateMutex.unlock();
 
     _wsHelper->quitWebSocketThread();
-    std::shared_ptr<bool> isDestroyed = _isDestroyed;
+    std::shared_ptr<std::atomic<bool>> isDestroyed = _isDestroyed;
     _wsHelper->sendMessageToCocosThread([this, isDestroyed](){
```

Comment:
So I create a new local variable `isDestroyed` shared_ptr to share the internal value of `WebSocket::_isDestroyed`.
Lambda will catch `isDestroyed` and get the shared value in `WebSocket::_isDestroyed` although `WebSocket::_isDestroyed` shared_ptr itself is destroyed. But we could catch its internal value.